### ISSUE TITLE
Edge daemon 1.1.6 creates workload sockets with incorrect owners leading to modules failing to start

### ISSUE THREAD

**Issue by corollaries** [Azure/iotedge#5693]
Recently we upgraded our devices to use iotedge 1.1.6 on a Yocto based OS. In some cases when a device starts for the first time, some of the modules that are being deployed are never properly created. 

The following logs were extracted from `edgeAgent`:

```
Oct 13 14:18:12 iotedge-device edgeAgent[677]: <6> 2021-10-13 12:18:12.342 +00:00 [INF] - Updated reported properties
Oct 13 14:18:20 iotedge-device edgeAgent[677]: <6> 2021-10-13 12:18:20.520 +00:00 [INF] - Module 'OperatorWeb' scheduled to restart after 05m:00s (04m:55s left).
Oct 13 14:18:20 iotedge-device edgeAgent[677]: <6> 2021-10-13 12:18:20.520 +00:00 [INF] - Plan execution started for deployment 66
Oct 13 14:18:20 iotedge-device edgeAgent[677]: <6> 2021-10-13 12:18:20.521 +00:00 [INF] - Executing command: "Command Group: (\n  [Create module CustomModule]\n  [Start module CustomModule]\n)"
Oct 13 14:18:20 iotedge-device edgeAgent[677]: <6> 2021-10-13 12:18:20.521 +00:00 [INF] - Executing command: "Create module CustomModule"
Oct 13 14:19:23 iotedge-device edgeAgent[677]: <3> 2021-10-13 12:19:23.457 +00:00 [ERR] - Executing command for operation ["create"] failed.
Oct 13 14:19:23 iotedge-device edgeAgent[677]: Microsoft.Azure.Devices.Edge.Agent.Edgelet.EdgeletCommunicationException- Message:Error calling Create module CustomModule: Could not create module CustomModule
Oct 13 14:19:23 iotedge-device edgeAgent[677]:         caused by: Could not get module CustomModule, StatusCode:500, at:   at Microsoft.Azure.Devices.Edge.Agent.Edgelet.Version_2020_07_07.ModuleManagementHttpClient.HandleException(Exception exception, String oper
ation) in /home/vsts/work/1/s/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2020_07_07/ModuleManagementHttpClient.cs:line 232
Oct 13 14:19:23 iotedge-device edgeAgent[677]:    at Microsoft.Azure.Devices.Edge.Agent.Edgelet.Versioning.ModuleManagementHttpClientVersioned.Execute[T](Func`1 func, String operation) in /home/vsts/work/1/s/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/
versioning/ModuleManagementHttpClientVersioned.cs:line 155
Oct 13 14:19:23 iotedge-device edgeAgent[677]:    at Microsoft.Azure.Devices.Edge.Agent.Edgelet.Version_2020_07_07.ModuleManagementHttpClient.CreateModuleAsync(ModuleSpec moduleSpec) in /home/vsts/work/1/s/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/ve
rsion_2020_07_07/ModuleManagementHttpClient.cs:line 102
Oct 13 14:19:23 iotedge-device edgeAgent[677]:    at Microsoft.Azure.Devices.Edge.Agent.Core.LoggingCommandFactory.LoggingCommand.ExecuteAsync(CancellationToken token) in /home/vsts/work/1/s/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/LoggingCommandFactor
y.cs:line 67
Oct 13 14:19:23 iotedge-device edgeAgent[677]: <3> 2021-10-13 12:19:23.459 +00:00 [ERR] - Executing command for operation ["Command Group: (\n  [Create module CustomModule]\n  [Start module CustomModule]\n)"] failed.
Oct 13 14:19:23 iotedge-device edgeAgent[677]: Microsoft.Azure.Devices.Edge.Agent.Edgelet.EdgeletCommunicationException- Message:Error calling Create module CustomModule: Could not create module CustomModule
Oct 13 14:19:23 iotedge-device edgeAgent[677]:         caused by: Could not get module CustomModule, StatusCode:500, at:   at Microsoft.Azure.Devices.Edge.Agent.Edgelet.Version_2020_07_07.ModuleManagementHttpClient.HandleException(Exception exception, String oper
ation) in /home/vsts/work/1/s/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2020_07_07/ModuleManagementHttpClient.cs:line 232
Oct 13 14:19:23 iotedge-device edgeAgent[677]:    at Microsoft.Azure.Devices.Edge.Agent.Edgelet.Versioning.ModuleManagementHttpClientVersioned.Execute[T](Func`1 func, String operation) in /home/vsts/work/1/s/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/
versioning/ModuleManagementHttpClientVersioned.cs:line 155
Oct 13 14:19:23 iotedge-device edgeAgent[677]:    at Microsoft.Azure.Devices.Edge.Agent.Edgelet.Version_2020_07_07.ModuleManagementHttpClient.CreateModuleAsync(ModuleSpec moduleSpec) in /home/vsts/work/1/s/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/ve
rsion_2020_07_07/ModuleManagementHttpClient.cs:line 102
```

Specifically the `StatusCode:500` received from the `edge-daemon` is troubling. The following logs were extracted from `edge-daemon`. (This was extracted a minute earlier but `edge-daemon` keeps logging this until it stops trying to run the deployment properly)

```
Oct 13 14:17:09 iotedge-device iotedged[748]: 2021-10-13T12:17:09Z [INFO] - Pulling image <REDACTED>...
Oct 13 14:17:09 iotedge-device iotedged[748]: 2021-10-13T12:17:09Z [INFO] - Successfully pulled image <REDACTED>
Oct 13 14:17:09 iotedge-device iotedged[748]: 2021-10-13T12:17:09Z [INFO] - Creating module CustomModule...
Oct 13 14:17:09 iotedge-device iotedged[748]: 2021-10-13T12:17:09Z [INFO] - Starting new listener for module CustomModule
Oct 13 14:17:09 iotedge-device iotedged[748]: [[0;1;31m[[0;1;39m[[0;1;31m2021-10-13T12:17:09Z [ERR!] - Internal server error: Could not create module CustomModule[[0m
Oct 13 14:17:09 iotedge-device iotedged[748]:         caused by: Could not get module CustomModule
```

Based on the releasenotes of the last few IoT Edge versions I managed to track down where the issue is. The following output is of `ls -la /var/lib/iotedge/mnt/` (These are not our true module names)
```
drwxr-xr-x 6 iotedge iotedge 4096 Oct 13 13:41 .
drwxr-xr-x 5 iotedge iotedge 4096 Oct 12 21:03 ..
drwxr-xr-x 2 root    root    4096 Oct 12 21:15 CustomModule1.sock
srw-rw-rw- 1 iotedge iotedge    0 Oct 13 11:29 CustomModule2.sock
drwxr-xr-x 2 root    root    4096 Oct 12 21:15 CustomModule.sock
srw-rw-rw- 1 iotedge iotedge    0 Oct 13 11:30 CustomModule3.sock
drwxr-xr-x 2 root    root    4096 Oct 12 21:15 CustomModule4.sock
srw-rw-rw- 1 iotedge iotedge    0 Oct 13 11:28 CustomModule5.sock
srw-rw-rw- 1 iotedge iotedge    0 Oct 13 11:26 CustomModule6.sock
srw-rw-rw- 1 iotedge iotedge    0 Oct 13 11:27 CustomModule7.sock
drwxr-xr-x 2 root    root    4096 Oct 12 21:15 CustomModule8.sock
srw-rw-rw- 1 iotedge iotedge    0 Oct 13 11:27 CustomModule9.sock
srw-rw-rw- 1 iotedge iotedge    0 Oct 13 11:28 CustomModule10.sock
srw-rw-rw- 1 iotedge iotedge    0 Oct 13 13:41 CustomModule11.sock
srw-rw-rw- 1 iotedge iotedge    0 Oct 13 11:28 CustomModule12.sock
srw-rw-rw- 1 iotedge iotedge    0 Oct 12 21:03 CustomModule13.sock
srw-rw-rw- 1 iotedge iotedge    0 Oct 12 21:05 CustomModule14.sock
```

The modules that failed to start had a workload socket created with an incorrect owner (`root` instead of `iotedge`) and is a directory instead of a file. Manually removing this file and redeploying makes this problem disappear. Removing this file is not a maintainable solution on our end so we are thinking of downgrading to `iotedge 1.1.3` to not have this problem. 

Is this issue known and is there a way we can mitigate this remotely without having to manually login and remove the file?
Unfortunately this issue is not easy for us to reproduce as it only seems to happen on devices that have never booted before.

### Device Information
* Custom Yocto OS
* 1.1.6
* Linux Containers

### Runtime Versions
* iotedge 1.1.6
* Edge Agent 1.1.5
* Edge Hub 1.1.5
* Docker 19.03.8

**Comment by huguesBouvier** [comment]
Hello, do you have specific logs (maybe from docker) that could explain why module creation failed (Could not create module CustomModule)?

**Comment by veyalla** [comment]
Errors related to this were fixed in 1.1.8 that was released yesterday. 1.2.5 release will follow shortly with same fixes. Closing this issue as it is referencing 1.1 branch.

**Comment by EnricoBeltramo** [comment]
I'm falling in a similar issue, but I'm using iotedge with Yocto (https://github.com/Azure/meta-iotedge) that is update only to version 1.1.6 when is planned to apply this update to yocto layer too?
